activerecord/lib/active_record/associations.rb

```diff
@@ -611,6 +616,9 @@ def association_instance_set(name, association)
       # Similarly, if any of the +before_remove+ callbacks throw an exception, the object
       # will not be removed from the collection.
       #
+      # Note: In order to trigger remove callbacks use either +firm.clients.destroy_all+ or +firm.clients.destroy(c).
+      # Attempting to remove them using +firm.clients.destroy+ will not trigger callbacks.
```

Comment:
This last sentence isn't accurate. `firm.clients.destroy` won't trigger callbacks because it doesn't work, it returns `nil`. Nothing gets deleted because an argument is required - `destroy` can't operate on more than one record. Did you mean to put `firm.clients.delete_all` or `firm.clients.delete(c)` here?

Reply:
@eileencodes Based on Nick's explanation in the original issue https://github.com/rails/rails/issues/14365#issuecomment-657292394 I think the last sentence was intentional, but as you said, using `destroy` doesn't operates on more than one record, maybe using `firm.clients.first.destroy` would be a better example? On the issue there were some cases (https://github.com/rails/rails/issues/14365#issuecomment-54271422 https://github.com/rails/rails/issues/14365#issuecomment-55131537 https://github.com/rails/rails/issues/14365#issuecomment-335942314 https://github.com/rails/rails/issues/14365#issuecomment-500185257) that folks used `.first.destroy`/`.last.destroy` which didn't dispatch the callback either.

I think the main confusion is with `destroy` since the docs for `delete_all` already explain that the callbacks aren't fired https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-delete_all

Reply:
> This last sentence isn't accurate IIRC. firm.clients.destroy won't trigger callbacks

yup thats correct, and my intention was indeed to tell users not to use `destroy` this way. 

> destroy can't operate on more than one record

yup also correct, but the way the method is defined in the collection proxy, it uses `*records` which if you don't pass anything to this will be an empty `[]` and we have a check that returns `nil` when it empty.

Reply:
ok sounds good. I will add `delete` and `delete_all` in the list. What do you think of changing the last sentence a bit to be more of a warning instead, since calling `destroy` without args will lead to records being removed, just no `after_remove` hooks fired. I think this was the main confusion in the original issue.